### PR TITLE
PeerInfo::Clear

### PR DESCRIPTION
**Pull request by Brett208**

### CODE REVIEW

File: client/OPUNetTransportLayer.h [hunk 1]
```diff
@@ -39,6 +39,13 @@ struct PeerInfo
 	PeerStatus status;
 	sockaddr_in address;
 	bool bReturnJoinPacket;
+
+	void Clear()
```
Comment (Brett208) [inline client/OPUNetTransportLayer.h:43]: Wasn't sure if we wanted to go ahead and move PeerInfo into its own header/cpp file and continue to expand its logic.

File: client/OPUNetTransportLayer.h [hunk 2]
```diff
@@ -39,6 +39,13 @@ struct PeerInfo
 	PeerStatus status;
 	sockaddr_in address;
 	bool bReturnJoinPacket;
```
Comment (Brett208) [inline client/OPUNetTransportLayer.h:41]: bReturnJoinPacket is set when PeerInfo was cleared in source code. I didn't know if this was intentional. I would like to set it to false when clearing, but didn't look at the source code close enough to make sure it wouldn't negatively affect logic.

Comment (DanRStevens) [reply client/OPUNetTransportLayer.h:41]: Yeah, I don't know. It sure looks intentional.

File: client/OPUNetTransportLayer.h (hunk 1 above)
Comment (DanRStevens) [reply client/OPUNetTransportLayer.h:43]: I think it's fine to leave it where it is for now. It's meant to be a struct with all public fields, and doesn't really stand useful on it's own, but only in the context of where it is used.

**Review by DanRStevens — APPROVED** [review]
Looks good. I like the range for change.

----

Was thinking, it might be possible to use aggregate initialization syntax to overwrite old struct values. That might provide an alternative to writing `Clear()` methods:
```cpp
someExistingStruct = {}; // Clear struct <-- with a useless comment :P
```

That would be most relevant when overwriting all values though, which I suspect is not necessarily the case here. I'm not too certain how it would behave if a non-zero default value was set for one of the fields. Might be worth experimenting with if you're curious. Probably a post merge thing to try out.

**Comment by Brett208** [comment]
Thanks for the review. I'd like to look into the aggregate initialization but until we figure out if the boolean value inside the PeerInfo struct can be set, probably best to leave alone.

I was thinking longer term of expanding PeerInfo to be more than a simple struct to move some logic out of OPUNetTransportLayer since the source code file is pretty long.

**Pull request merged**